global_helpers/panther_base_helpers.py

```diff
@@ -494,3 +495,31 @@ def m365_alert_context(event):
 def defang_ioc(ioc):
     """return defanged IOC from 1.1.1.1 to 1[.]1[.]1[.]1"""
     return ioc.replace(".", "[.]")
+
+
+def unmock(func: callable, *args, **kwargs):
+    """Wraps mocked functions to perform automatic JSON unmarshalling on the returned value during
+    unit tests, and return the unmodified value during actual log analysis.
+
+    Usage:
+        func: the mocked function you want to call
+        *args: the positional arguments of 'func', if any
+        **kwargs: the keyword arguments of 'func', if any
+
+    Example:
+        To wrap `get_counter("cache-key", force_ttl=True)`, call:
+            unmock(get_counter, "cache-key", force_ttl=True)
```

Comment:
Could we possible have this return a wrapper instead? That way we could say
```
from thing import get_counter
from mocking import unmock

get_counter = unmock(get_counter)
```
So all mocking logic could live at the top of the file and the rule logic would look no different

Reply:
I don’t know if this would work, but just a thought

Reply:
Oh maybe! I'm not sure when the mock replacement happens - it might replace the existing function if we do the override at the module level. Can test it though - this flow is much cleaner, so if I can get it to work, that's be sweet

Reply:
Just tested it - the mocking replacement happens after the module is imported, so anything we override at the module level gets replaced with the mock function. We could advise customers to make the replacement at the top of their `rule` function, but they'd need to remember to do the same if they wanna use it in any auxilliary functions too. What do you think is the simpler implementation?